instrumentation-tools/src/api_surface.py

```diff
@@ -2,15 +2,18 @@
 #  This product includes software developed at Datadog (https://www.datadoghq.com/).
 #  Copyright 2016-Present Datadog, Inc.
 
-import re
+from builtins import *
 from math import floor
+import re
 
-DIVIDER = "#"
 DOT = "."
+NESTED_TYPE_SEPARATOR = "$"
+API_SEPARATOR = "#"
 EXTEND_SEPARATOR = ":"
-TESTABLE_API_REGEX = re.compile("^(?<!deprecated)(\\s*)(constructor|fun)(.+)", re.IGNORECASE)
-TYPE_REGEX = re.compile("(\\s*)(.*)(class|interface|enum|object)(\\s)(.+)$", re.IGNORECASE)
-DEFAULT_IGNORED_TYPE_REGEX = re.compile("(.*)(companion|deprecated)(\\s)(.*)$", re.IGNORECASE)
+TESTABLE_API_REGEX = re.compile("^(\\s*)(?<!deprecated)(\\s*)(constructor|fun)(.+)")
+TYPE_REGEX = re.compile("(\\s*)(.*)(class|interface|enum|object)(\\s)(.*)$")
```

Comment:
```suggestion
TYPE_REGEX = re.compile("^(\\s*)(.*)(class|interface|enum|object)(\\s)(.+)$")
```

Reply:
This `*` is on purpose in order to catch `companion object` declarations.

```
// note there's a trailing space in the following line
      companion object 
(\s*)|  (.*)   |object|(\s)|(.*)
```